src/coreclr/debug/dbgutil/machoreader.cpp

```diff
@@ -207,11 +207,11 @@ MachOModule::ReadLoadCommands()
                 m_segments.push_back(segment);
 
                 // Calculate the load bias for the module. This is the value to add to the vmaddr of a
-                // segment to get the actual address. 
-                if (strcmp(segment->segname, SEG_TEXT) == 0)
+                // segment to get the actual address. For shared modules, this is 0 since those segments
+                // are absolute address.
+                if (segment->fileoff == 0 && segment->filesize > 0)
```

Comment:
I am just curious - is this a simplification or was the previous way incorrect?

Reply:
The previous way only worked correctly for "user" modules (base address < 0x00007FFF00000000) and not for "shared" or system modules (> that adresss).

Reply:
It also indicates that the module's segments are already absolute address (load bias = 0) or relative (load bias > 0).